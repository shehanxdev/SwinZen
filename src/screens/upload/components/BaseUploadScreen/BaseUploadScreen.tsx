import React, { PropsWithChildren } from 'react';
import { ImageBackground, SafeAreaView, ScrollView } from 'react-native';

import { images } from '@sz/assets';
import { tw } from '@sz/config';
import { useHeaderHeight } from '@sz/hooks';

export interface BaseUploadScreenProps extends PropsWithChildren {
  testID?: string;
  isSetupScreen?: boolean;
}

export function BaseUploadScreen({
  children,
  testID = 'BaseUploadScreenTestID',
  isSetupScreen = false,
}: BaseUploadScreenProps) {
  const headerHeight = useHeaderHeight();
  const backgroundStyle = isSetupScreen ? tw`flex-1` : tw`h-full pt-[${headerHeight}px bg-Neutral-Sz900`;

  return (
    <ImageBackground
      testID={testID}
      style={backgroundStyle}
      source={isSetupScreen ? images.videoSetupBackground : undefined}>
      <SafeAreaView style={tw`h-full pt-[${headerHeight}px]`}>
        <ScrollView contentContainerStyle={tw`grow`}>{children}</ScrollView>
      </SafeAreaView>
    </ImageBackground>
  );
}
