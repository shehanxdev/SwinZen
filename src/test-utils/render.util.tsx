import { NavigationContext } from '@react-navigation/native';
import { render } from '@testing-library/react-native';
import React, { ReactNode } from 'react';
import Config from 'react-native-config';
import { Provider as PaperProvider } from 'react-native-paper';
import { Provider } from 'react-redux';

import { paperTheme } from '@sz/config';
import { ConfigService } from '@sz/services';
import { store } from '@sz/stores';

type RenderParams = Parameters<typeof render>;

export function renderWithProviders(ui: RenderParams[0], options?: RenderParams[1]) {
  ConfigService.setCallback(key => Config[key]);

  const BaseProviders = ({ children }: { children: ReactNode }) => (
    <Provider store={store}>
      <NavigationContext.Provider
        //@ts-ignore
        value={{
          isFocused: () => true,
          addListener: jest.fn(() => jest.fn()),
        }}>
        <PaperProvider theme={paperTheme}>{children}</PaperProvider>
      </NavigationContext.Provider>
    </Provider>
  );

  return render(ui, { wrapper: BaseProviders, ...options });
}
