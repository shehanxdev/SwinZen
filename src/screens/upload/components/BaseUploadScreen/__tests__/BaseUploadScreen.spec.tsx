import * as React from 'react';

import { renderWithProviders } from '@sz/test-utils';

import { BaseUploadScreen, BaseUploadScreenProps } from '../BaseUploadScreen';

describe('BaseUploadScreen Component', () => {
  const testID = 'BaseUploadScreenTestID';
  const getRenderedScreen = (props?: Partial<BaseUploadScreenProps>) =>
    renderWithProviders(
      <BaseUploadScreen {...props}>
        <></>
      </BaseUploadScreen>,
    );

  it(`should render BaseUploadScreen component correctly`, () => {
    const rendered = getRenderedScreen();
    const renderedTree = rendered.toJSON();
    expect(renderedTree).toMatchSnapshot();
  });

  it(`should find the BaseUploadScreen component via testID`, () => {
    const { getByTestId } = getRenderedScreen();
    const foundCustomHeaderComponent = getByTestId(testID);
    expect(foundCustomHeaderComponent).toBeTruthy();
  });
});
