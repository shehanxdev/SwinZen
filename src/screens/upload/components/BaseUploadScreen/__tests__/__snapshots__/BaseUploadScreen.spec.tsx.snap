// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseUploadScreen Component should render BaseUploadScreen component correctly 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      {
        "backgroundColor": "#212529",
        "height": "100%",
        "paddingTop": 50,
      }
    }
  >
    <Image
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "height": "100%",
            "width": undefined,
          },
          undefined,
        ]
      }
      testID="BaseUploadScreenTestID"
    />
    <RCTSafeAreaView
      style={
        {
          "height": "100%",
          "paddingTop": 50,
        }
      }
    >
      <RCTScrollView
        contentContainerStyle={
          {
            "flexGrow": 1,
          }
        }
      >
        <View />
      </RCTScrollView>
    </RCTSafeAreaView>
  </View>
</View>
`;
